test/server/listener_manager_impl_test.cc

```diff
@@ -829,75 +828,40 @@ filter_chains: {}
   manager_->startWorkers(guard_dog_);
 
   EXPECT_EQ(0, server_.stats_store_.counter("listener_manager.listener_create_success").value());
+  checkStats(__LINE__, 1, 0, 0, 0, 1, 0, 0);
 
-  // TODO(lambdai): No need to invoke `addListenerToWorkerForTest` explicitly when intelligent warm
-  // up procedure is added.
-  ListenerImpl* listener_impl = dynamic_cast<ListenerImpl*>(listener_config);
-  auto overridden_listener = absl::make_optional<uint64_t>(1);
-  EXPECT_CALL(*worker_, addListener(_, _, _))
-      .WillOnce(Invoke([](absl::optional<uint64_t>, Network::ListenerConfig&,
-                          auto completion) -> void { completion(true); }));
-  manager_->addListenerToWorkerForTest(*worker_, overridden_listener, *listener_impl, nullptr);
-
-  EXPECT_EQ(1, server_.stats_store_.counter("listener_manager.listener_create_success").value());
-  EXPECT_CALL(*listener_foo, onDestroy());
-}
-
-TEST_F(ListenerManagerImplTest, DrainFilterChains) {
```

Comment:
This test is deleted because drainFilterChainsForTest() is retired in this PR. 
`DrainFilterChains` and `OverrideListener` is merged to the latter. The idea is that we can now drive the test flow by providing a listener config instead of the retired "ForTest" methods